shm/Observer.h

```diff
@@ -0,0 +1,7 @@
+#pragma once
+
+class Observer {
+public:
+    virtual ~Observer(){};
+    virtual void update() = 0;
+};
```

Comment:
Not as standalone class. You can have more Observer in future and have conflict with names